api/test/unit/conftest.py

```diff
@@ -129,8 +159,16 @@ def cleanup_elasticsearch_test_documents(request, settings):
 
     es: Elasticsearch = settings.ES
 
+    # If pook was activated by a test and not deactivated
+    # (usually because the test failed and something prevent
+    # pook from cleaning up after itself), disable here so that
```

Comment:
I didn't know there was a possibility that pook could not clean up. That's quite bad for the isolation of the tests.

Reply:
I've only seen it in some cases, and I wonder if it has to do with async stuff, but yes, I have seen this particular factory run and have errors logged at the end of the test run because pook was still intercepting requests. IIRC it was only on tests that relied on `pook.use` as a context manager, where the code inside the context raised an exception, so there may be some kind of bug in pook there.